projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php

```diff
@@ -2592,8 +2592,19 @@ public static function parse( $attributes, $content ) {
 			wp_enqueue_style( 'grunion.css' );
 		}
 
+		// Move align tool classes to the top container
+		$has_alignwide   = preg_match( '/<div class=\"wp-block-jetpack-contact-form.*alignwide.*?">/m', $content );
+		$has_alignfull   = preg_match( '/<div class=\"wp-block-jetpack-contact-form.*alignfull.*?">/m', $content );
+		$wrapper_classes = '';
+
+		if ( $has_alignwide ) {
+			$wrapper_classes = 'alignwide';
+		} elseif ( $has_alignfull ) {
+			$wrapper_classes = 'alignfull';
+		}
+
 		$r  = '';
-		$r .= "<div id='contact-form-$id'>\n";
+		$r .= "<div id='contact-form-$id' class='$wrapper_classes'>\n";
```

Comment:
Have you tried to add the `alignwide` and `alignfull` classes using the block `supports` directly?

For instance, when we add the `supports` array when the block is registered in `register_contact_form_blocks()`, we can then access it through `apply_block_supports()`:

```diff
diff --git a/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php b/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
index 099a38e1ed..a636271f78 100644
--- a/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
+++ b/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
@@ -398,6 +398,9 @@ class Grunion_Contact_Form_Plugin {
 			'jetpack/contact-form',
 			array(
 				'render_callback' => array( __CLASS__, 'gutenblock_render_form' ),
+				'supports'        => array(
+					'align' => array( 'wide', 'full' ),
+				),
 			)
 		);
 
@@ -2597,6 +2600,9 @@ class Grunion_Contact_Form extends Crunion_Contact_Form_Shortcode {
 		$has_alignfull   = preg_match( '/<div class=\"wp-block-jetpack-contact-form.*alignfull.*?">/m', $content );
 		$wrapper_classes = '';
 
+		$wrapper_attributes = \WP_Block_Supports::get_instance()->apply_block_supports();
+		var_dump( $wrapper_attributes );
+
 		if ( $has_alignwide ) {
 			$wrapper_classes = 'alignwide';
 		} elseif ( $has_alignfull ) {

```

In the example above we will get the following var dump:

```
array(1) { ["class"]=> string(44) "alignfull wp-block-jetpack-contact-form" }
```

where `alignfull` is coming directly from the block `support` and the alignment user selected in the editor. We did something similar in this PR: https://github.com/Automattic/jetpack/pull/22622.

This way we don't need to rely on `preg_match()`.

With that said, I might be missing something that you had in mind when you worked on this PR.